src/pkg/oci/utils.go

```diff
@@ -93,33 +99,46 @@ func (o *OrasRemote) FetchLayer(desc ocispec.Descriptor) (bytes []byte, err erro
 
 // FetchZarfYAML fetches the zarf.yaml file from the remote repository.
 func (o *OrasRemote) FetchZarfYAML(manifest *ZarfOCIManifest) (pkg types.ZarfPackage, err error) {
-	zarfYamlDescriptor := manifest.Locate(config.ZarfYAML)
-	if zarfYamlDescriptor.Digest == "" {
-		return pkg, fmt.Errorf("unable to find %s in the manifest", config.ZarfYAML)
+	return FetchYAML[types.ZarfPackage](o.FetchLayer, manifest, config.ZarfYAML)
+}
+
+// FetchImagesIndex fetches the images/index.json file from the remote repository.
+func (o *OrasRemote) FetchImagesIndex(manifest *ZarfOCIManifest) (index *ocispec.Index, err error) {
+	return FetchJSON[*ocispec.Index](o.FetchLayer, manifest, manifest.indexPath)
+}
+
+// FetchJSON fetches the given JSON file from the remote repository.
+func FetchJSON[T any](fetcher func(desc ocispec.Descriptor) (bytes []byte, err error), manifest *ZarfOCIManifest, path string) (result T, err error) {
```

Comment:
```go
// FetchJSON fetches the given JSON file from the remote repository.
func FetchJSON[T any](fetcher func(desc ocispec.Descriptor) (bytes []byte, err error), manifest *ZarfOCIManifest, path string) (result T, err error) {
	return FetchUnmarshal[T](fetcher, json.Unmarshal, manifest, path)
}

// FetchYAML fetches the given JSON file from the remote repository.
func FetchYAML[T any](fetcher func(desc ocispec.Descriptor) (bytes []byte, err error), manifest *ZarfOCIManifest, path string) (result T, err error) {
	return FetchUnmarshal[T](fetcher, goyaml.Unmarshal, manifest, path)
}

// FetchUnmarshal fetches the given descriptor from the remote repository and unmarshals it.
func FetchUnmarshal[T any](fetcher func(desc ocispec.Descriptor) (bytes []byte, err error), unmarshaler func(data []byte, v interface{}) error, manifest *ZarfOCIManifest, path string) (result T, err error) {
	descriptor := manifest.Locate(path)
	if IsEmptyDescriptor(descriptor) {
		return result, fmt.Errorf("unable to find %s in the manifest", path)
	}
	bytes, err := fetcher(descriptor)
	if err != nil {
		return result, err
	}
	err = unmarshaler(bytes, &result)
	if err != nil {
		return result, err
	}
	return result, nil
}
```
If you wanna be real cool

Reply:
I wasn't sure if this level of abstraction was worth doing, sacrificing readability even more than what exists. I will think about it.